invenio_rdm_records/resources/serializers/csl/__init__.py

```diff
@@ -18,3 +36,89 @@ class CSLJSONSerializer(MarshmallowJSONSerializer):
     def __init__(self, **options):
         """Constructor."""
         super().__init__(schema_cls=CSLJSONSchema, **options)
+
+
+class StringCitationSerializer(CSLJSONSerializer):
+    """CSL Citation Formatter serializer for records.
+
+    In order to produce a formatted citation of a record through citeproc-py,
+    we need a CSL-JSON serialized version of it.
+    """
+
+    _default_style = 'harvard1'
+    """The `citeproc-py` library supports by default the 'harvard1' style."""
+
+    _default_locale = 'en-US'
+    """The `citeproc-py` library supports by default the 'harvard1' style."""
+
+    _user_args = {
+        'style': fields.Str(missing=_default_style),
+        'locale': fields.Str(missing=_default_locale)
+    }
+    """Arguments for the webargs parser."""
+
+    _valid_formats = ('csl', 'bibtex')
+    """Supported formats by citeproc-py."""
+
+    def __init__(self, record_format='csl'):
+        """Initialize the inner record serializer.
+
+        :param record_format: Format that the serializer produces.
+        """
+        super().__init__()
+        self.record_format = record_format
+
+    @classmethod
+    def _get_args(cls, **kwargs):
+        """Parse style and locale.
+
+        Argument location precedence: kwargs > view_args > query
+        """
+        csl_args = {
+            'style': cls._default_style,
+            'locale': cls._default_locale
+        }
+
+        if has_request_context():
```

Comment:
I am wondering if we can avoid using `request` and many other `flask` objects... and use the `resource_requestctx`... this would require adding some parameters in the config...  e.g. in the new `resources` paradigm we already implemented an arg parser and here we are using `FlaskParser`, is a bit contradictory.



Maybe it can be improved in a different PR. @ntarocco this is maybe a discussion point for later on.

Reply:
I am not sure if it can be really improved with more params in the config?
At some point, the `view_args` needs to be parsed...

I would remove `if has_request_context` because here we are at the resource layer, I think that we are sure that we are in a request context. WDYT?
Question: why using `FlaskParser` and not simply `request.args.get("style")`? Maybe it was in Zenodo code?

Reply:
> Not need to check context...

I fully agree, in this layer we can assume we are in a request context.

> About parsing view_args

Maybe the parsing itself is needed and cannot be done in the same fashion than other args, I agree. However, we should tyr to be consistent and access it through the [`resource_requestctx.view_args`](https://github.com/inveniosoftware/flask-resources/blob/5ca81aa51715f2bdd12be25c2e71d6c6fb1d7671/flask_resources/context.py#L61)

Reply:
Ok, I will leave this for now.